solidity/contracts/vault/TBTCVault.sol

```diff
@@ -145,23 +145,47 @@ contract TBTCVault is IVault, Governable {
         _unmint(msg.sender, amount);
     }
 
-    /// @notice Burns `amount` of TBTC from the caller's account and transfers
-    ///         `amount` back to the caller's balance in the Bank.
-    /// @dev This function is doing the same as `unmint` but it allows to
-    ///      execute unminting without an additional approval transaction.
-    ///      The function can be called only via `approveAndCall` of TBTC token.
+    /// @notice Burns `amount` of TBTC from the caller's balance and transfers
+    ///         `amount` of Bank balance to the Bridge requesting redemption
+    ///         based on the provided `redemptionData`.
+    /// @param amount Amount of TBTC to unmint and request to redeem in Bridge.
+    /// @param redemptionData Redemption data in a format expected from
+    ///        `redemptionData` parameter of Bridge's `receiveBalanceApproval`
+    ///        function.
+    function redeem(uint256 amount, bytes calldata redemptionData) external {
```

Comment:
I think it does not because *unminting* is "TBTC to Bank balance" and *redemption* is "Bank balance to BTC". In this case, *unmint* has to happen before *redeem* and we are doing internally.

But... I see your point. I think `unmintAndApprove` is too cryptic. But what about `unmintAndRedeem`?

Reply:
See e12b1cf143b7003420b83df4a49a84f1bcd06a5d please.